crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -3,14 +3,114 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use syn::{parse::ParseStream, Attribute, Ident, Result, Token, Type};
+use std::collections::HashSet;
+
+use syn::{
+    parse::{Error, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    Attribute, Expr, Ident, Meta, MetaNameValue, Path, Result, Token, Type,
+};
+
+use quote::format_ident;
+
+/// An optional identifier of the functions passed with flags
+/// e.g. read = my_getter, IdentFlag would be used to store my_getter
+type IdentFlag = Option<Ident>;
+
+/// Struct for storing the flags provided for a QProperty and their optional identifiers ([IdentFlag])
+#[derive(Debug)]
+pub struct QPropertyFlags {
+    read: IdentFlag, // TODO: maybe change this to better represent the data
+    write: Option<IdentFlag>,
+    notify: Option<IdentFlag>,
+}
+
+impl QPropertyFlags {
+    pub fn new() -> Self {
+        Self {
+            read: None,
+            write: None,
+            notify: None,
+        }
+    }
```

Comment:
could instead just have `#[derive(Default)]` on the `QPropertyFlags` then you can use `QPropertyFlags::default()`, but then i guess maybe this is not the "default" this is none of them

Reply:
changed